boards/xtensa/esp32/doc/esp32.rst

```diff
@@ -0,0 +1,159 @@
+
+.. code-block:: console
+
+   export ZEPHYR_GCC_VARIANT="espressif"
+   export ESP_IDF_PATH=/path/to/esp-idf
+   export ESPRESSIF_TOOLCHAIN_PATH=/path/to/xtensa-esp32-elf/
+
+Flashing
+========
+
+Issuing ```make BOARD=esp32 flash``` should work as usual.  Environment
+variables can be set to set the serial port device, baud rate, and
+other options.  Please refer to the following table for details.
+
++----------------+---------------+
+| Variable       | Default value |
++================+===============+
+| ESP_DEVICE     | /dev/ttyUSB0  |
+| ESP_BAUD_RATE  | 921600        |
+| ESP_FLASH_SIZE | detect        |
+| ESP_FLASH_FREQ | 40m           |
+| ESP_FLASH_MODE | dio           |
+| ESP_TOOL       | espidf        |
++----------------+---------------+
+
+It's impossible to determine which serial port the ESP32 board is
+connected to, as it uses a generic RS232-USB converter.  The default of
+```/dev/ttyUSB0``` is provided as that's often the assigned name on a Linux
+machine without any other such converters.
+
+The baud rate of 921600bps is recommended.  If experiencing issues when
+flashing, try halving the value a few times (460800, 230400, 115200,
+etc).  It might be necessary to change the flash frequency or the flash
+mode; please refer to the esptool documentation for guidance on these
+settings.
+
+If ```ESP_TOOL``` is set to "espidf", the esptool.py script found within
+ESP-IDF will be used.  Otherwise, this variable is handled as a path to
+the tool.
+
+Using JTAG
+==========
+
+In addition to the non-official Zephyr SDK, this board also requires a build
+of OpenOCD that has been patched by the SoC vendor.  The source code and build
+instructions can be obtained from GitHub:
+
+.. code-block:: console
+
+   git clone https://github.com/espressif/openocd-esp32
+
+After building OpenOCD, one should proceed as usual with the set up process. 
+Since JTAG adapters may be different, this will most likely require editing
+the ```esp32.cfg``` file that's in the ```openocd-esp32``` directory.  For
+instance, to use a Flyswatter 2 JTAG adapter, one would modify the file so
+that the "source" line would read:
+
+.. code-block:: tcl
+
+   source [find interface/ftdi/flyswatter2.cfg]
+
+It might be a good idea to comment the line setting ```ESP32_RTOS``` and
+increasing ```adapter_khz``` to 400.
+
+After the file has been properly edited, connect the JTAG pins according to
+the table below.  Please consult your JTAG adapter manual for the proper
+pinout.  Power to the ESP32 board should be provided by a USB cable.
+
++------------+-----------+
+| ESP32 pin  | JTAG pin  |
++============+===========+
+| 3V3        | VTRef     |
+| EN         | nTRST     |
+| IO14       | TMS       |
+| IO12       | TDI       |
+| GND        | GND       |
+| IO13       | TCK       |
+| IO15       | TDO       |
++------------+-----------+
+
+Further documentation can be obtained from the SoC vendor in `JTAG debugging
+for ESP32`_.
+
+References
+**********
+
```

Comment:
add the footnote referenced by the first paragraph;

     .. [1] https://en.wikipedia.org/wiki/ESP32